backend/app/routers/datasets.py

```diff
@@ -509,6 +513,61 @@ async def save_files(
         raise HTTPException(status_code=404, detail=f"Dataset {dataset_id} not found")
 
 
+@router.post("/{dataset_id}/local_files", response_model=FileOut)
+async def save_local_file(
+    localfile_in: LocalFileIn,
+    dataset_id: str,
+    folder_id: Optional[str] = None,
+    user=Depends(get_current_user),
+    es=Depends(dependencies.get_elasticsearchclient),
+    rabbitmq_client: BlockingChannel = Depends(dependencies.get_rabbitmq),
+    allow: bool = Depends(Authorization("uploader")),
+):
+    if (dataset := await DatasetDB.get(PydanticObjectId(dataset_id))) is not None:
+        if user is None:
+            raise HTTPException(
+                status_code=401, detail=f"User not found. Session might have expired."
+            )
+
+        # Check to make sure file is cleared to proceed
+        cleared = False
+        for wpath in settings.LOCAL_WHITELIST:
+            if localfile_in.path.startswith(wpath):
+                cleared = True
+        if not cleared:
+            raise HTTPException(
+                status_code=401,
```

Comment:
Should this be a 500 since it's an internal error? Or something less generic? 401 confused be as unauthorized.

Reply:
changed this to 500